components/selectors/matching.rs

```diff
@@ -466,7 +452,10 @@ fn matches_simple_selector<E>(
         }
         SimpleSelector::Negation(ref negated) => {
             !negated.iter().all(|s| {
-                matches_complex_selector(s, element, parent_bf, relations, flags)
+                match matches_complex_selector_internal(s, element, relations, flags) {
```

Comment:
(I meant that because `:not` is already quite expensive, but sure, will do that).